sweep/fee_bumper.go

```diff
@@ -673,20 +687,29 @@ func (t *TxPublisher) notifyResult(result *BumpResult) {
 func (t *TxPublisher) removeResult(result *BumpResult) {
 	id := result.requestID
 
-	// Remove the record from the maps if there's an error. This means this
-	// tx has failed its broadcast and cannot be retried. There are two
-	// cases,
+	var txid chainhash.Hash
+	if result.Tx != nil {
+		txid = result.Tx.TxHash()
+	}
+
+	// Remove the record from the maps if there's an error or the tx is
+	// confirmed. When there's an error, it means this tx has failed its
+	// broadcast and cannot be retried. There are two cases,
 	// - when the budget cannot cover the fee.
-	// - when a non-RBF related error occurs.
+	// - when a non-fee related error returned from PublishTransaction.
 	switch result.Event {
 	case TxFailed:
 		log.Errorf("Removing monitor record=%v, tx=%v, due to err: %v",
-			id, result.Tx.TxHash(), result.Err)
+			id, txid, result.Err)
 
 	case TxConfirmed:
-		// Remove the record is the tx is confirmed.
+		// Remove the record if the tx is confirmed.
 		log.Debugf("Removing confirmed monitor record=%v, tx=%v", id,
-			result.Tx.TxHash())
+			txid)
+
+	case TxError:
+		// Remove the record if there's an error.
+		log.Debugf("Removing monitor record=%v due to error", id)
```

Comment:
will `result.Error` not be populated in this case & ie can be included in log?

Reply:
yeah it should always be populated and now it's included in the log.